src/app/shared/components/button/button.component.html

```diff
@@ -1,4 +1,7 @@
-<div class="app-button">
-  <!-- Replace the text below with your code -->
-  Button
+<div
+  class="app-button"
+  [ngClass]="iconName && !buttonText ? 'app-button__single-icon' : ''"
```

Comment:
Technically this is fine, I would only highlight another syntax that can be used: `[ngClass]="{'app-button__single-icon': iconName && !buttonText}"`
You can spot the difference there, the class is applied only when the condition evaluates to true, not the other way around (check the condition and add a class, for false in  this case nothing)